addons/binding/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/handler/Enigma2Handler.java

```diff
@@ -0,0 +1,254 @@
+                if (command instanceof StringType) {
+                    commandExecutor.sendWarning((StringType) command);
+                } else {
+                    logger.warn("Invalid command type: {}: {}", command.getClass(), command);
+                }
+                break;
+            case CHANNEL_SEND_QUESTION:
+                if (command instanceof StringType) {
+                    commandExecutor.sendQuestion((StringType) command);
+                } else {
+                    logger.warn("Invalid command type: {}: {}", command.getClass(), command);
+                }
+                break;
+            default:
+                logger.warn("{} : Got command '{}' for channel '{}' which is unhandled!", command, channelUID.getId());
+                break;
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        refresher.removeListener(this);
+    }
+
+    @Override
+    public void getUpdate() {
+        OnOffType powerState = (OnOffType) commandExecutor.getPowerState();
+        if (powerState != null) {
+            updateState(CHANNEL_POWER, powerState);
+            if (powerState == OnOffType.ON) {
+                if (commandExecutor.getVolumeState() != null) {
+                    updateState(CHANNEL_VOLUME, commandExecutor.getVolumeState());
+                }
+                if (commandExecutor.getMutedState() != null) {
+                    updateState(CHANNEL_MUTE, commandExecutor.getMutedState());
+                }
+                if (commandExecutor.getChannelState() != null) {
+                    updateState(CHANNEL_CHANNEL, commandExecutor.getChannelState());
+                }
+                if (commandExecutor.getAnswerState() != null) {
+                    updateState(CHANNEL_GET_ANSWER, commandExecutor.getAnswerState());
+                }
+                if (commandExecutor.getNowPlayingTitle() != null) {
+                    updateState(CHANNEL_NOW_PLAYING_TITLE, commandExecutor.getNowPlayingTitle());
+                }
+                if (commandExecutor.getNowPlayingDescription() != null) {
+                    updateState(CHANNEL_NOW_PLAYING_DESCRIPTION, commandExecutor.getNowPlayingDescription());
+                }
+                if (commandExecutor.getNowPlayingDescriptionExtended() != null) {
+                    updateState(CHANNEL_NOW_PLAYING_DESCRIPTION_EXTENDED,
+                            commandExecutor.getNowPlayingDescriptionExtended());
+                }
+            } else {
+                updateState(CHANNEL_CHANNEL, new StringType(""));
+                updateState(CHANNEL_NOW_PLAYING_TITLE, new StringType("-"));
+                updateState(CHANNEL_NOW_PLAYING_DESCRIPTION, new StringType("-"));
+                updateState(CHANNEL_NOW_PLAYING_DESCRIPTION_EXTENDED, new StringType("-"));
+            }
+        }
+    }
+
+    public String getUserName() {
+        return (String) thing.getConfiguration().get(DEVICE_PARAMETER_USER);

```

Comment:
Looking at the current (since ~1 year) ESH sources, I really don't see how this happens. So as soon as you get a clue how they get in there, I'd be happy to have a look.

> i get a classcast exception, "java.math.BigDecimal cannot be cast to java.lang.Long". Althought i definitly add a Long to the config map

That actually sounds the other way round! Stuffing longs into the config, getting a CCE when trying to read it as long. In that case I'd say: the normalization works perfectly fine 😉